packages/core/echo/echo-db/src/api.test.ts

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2025 DXOS.org
+//
+
+import { Schema as S } from 'effect';
+import { describe, test } from 'vitest';
+
+import { raise } from '@dxos/debug';
+import {
+  Echo,
+  EntityKind,
+  FormatEnum,
+  FormatAnnotation,
+  getObjectAnnotation,
+  getSchema,
+  getSchemaDXN,
+  getSchemaTypename,
+  getSchemaVersion,
+  isInstanceOf,
+} from '@dxos/echo-schema';
+import { create, makeRef } from '@dxos/live-object';
+
+// TODO(dmaretskyi): Do all ECHO api's go into `Echo` or do some things like `create` and `Ref` stay separate?
+//  RB: Let's get the main schema types cleanup up first since that has the biggest impact.
+
+// RB: Commented out since lint error (is this just for debugging?)
+// This odd construct only serves one purpose: when you hover over `const x: Live<T>` you'd see `Live<T>` type.
+// interface _Live<T> {}
+// type Live<T> = _Live<T> & T;
+// const create = create_ as <T>(schema: S.Schema<T>, obj: T, meta?: ObjectMeta) => Live<T>;
+
+// RB: I prefer this format since it seems like the most natural extension to the default effect API; and therefore, the most incremental.
+const Org = S.Struct({
+  name: S.String,
+}).pipe(
+  Echo.Type({
+    typename: 'example.com/type/Org',
+    version: '0.1.0',
+  }),
+);
+
+// TODO(burdon): Remove Schema/Type suffix in Composer.
+interface Org extends S.Schema.Type<typeof Org> {}
+
+const Contact = S.Struct({
+  name: S.String,
+  dob: S.optional(S.String),
+  email: S.optional(S.String.pipe(FormatAnnotation.set(FormatEnum.Email))),
+  org: S.optional(Echo.Ref(Org)),
+}).pipe(
+  Echo.Type({
+    typename: 'example.com/type/Contact',
+    version: '0.1.0',
+  }),
+);
+
+interface Contact extends S.Schema.Type<typeof Contact> {}
+
+const Message = S.Struct({
+  // TODO(burdon): Support S.Date; Custom Timestamp (with defaults).
+  // TODO(burdon): Support defaults (update create and createStatic).
```

Comment:
Defaults should be applied on read, not on write, to support schema migrations with new properties and default values.

Reply:
i think defaults should be applied when the object is constructed -- and when read from disk (for schema updates).